diff --git a/node_modules/astro/dist/core/config/vite-load.js b/node_modules/astro/dist/core/config/vite-load.js
index 372821b..06c9ec9 100644
--- a/node_modules/astro/dist/core/config/vite-load.js
+++ b/node_modules/astro/dist/core/config/vite-load.js
@@ -2,7 +2,10 @@ import { pathToFileURL } from "node:url";
 import { createServer } from "vite";
 import loadFallbackPlugin from "../../vite-plugin-load-fallback/index.js";
 import { debug } from "../logger/core.js";
+import configAliasVitePlugin from "../../vite-plugin-config-alias/index.js";
+import { loadTSConfig } from "./tsconfig.js";
 async function createViteServer(root, fs) {
+  const tsConfig = await loadTSConfig(root);
   const viteServer = await createServer({
     configFile: false,
     server: { middlewareMode: true, hmr: false, watch: null, ws: false },
@@ -10,7 +13,13 @@ async function createViteServer(root, fs) {
     clearScreen: false,
     appType: "custom",
     ssr: { external: true },
-    plugins: [loadFallbackPlugin({ fs, root: pathToFileURL(root) })]
+    plugins: [
+      loadFallbackPlugin({ fs, root: pathToFileURL(root) }),
+      configAliasVitePlugin({ settings: {
+        tsConfig: tsConfig.tsconfig,
+        tsConfigPath: tsConfig.tsconfigFile
+      } })
+    ]
   });
   return viteServer;
 }
